src/bit-systems/network-debug.ts

```diff
@@ -0,0 +1,47 @@
+import { defineQuery, enterQuery, exitQuery, removeEntity } from "bitecs";
+import { Material, Scene, Vector3 } from "three";
+import { Text as TroikaText } from "troika-three-text";
+import { HubsWorld } from "../app";
+import { Networked } from "../bit-components";
+import { NetworkDebugPrefab } from "../prefabs/network-debug";
+import { formatComponentProps, formatObjectName } from "../react-components/debug-panel/ECSSidebar";
+import { renderAsEntity } from "../utils/jsx-entity";
+import { forEachMaterial } from "../utils/material-utils";
+import { EntityID } from "../utils/networking-types";
+
+const texts = new Map<EntityID, TroikaText>();
+const networkedQuery = defineQuery([Networked]);
+const enteredNetworkedQuery = enterQuery(networkedQuery);
+const exitedNetworkedQuery = exitQuery(networkedQuery);
+const _scale = new Vector3();
+export function networkDebugSystem(world: HubsWorld, scene: Scene) {
+  enteredNetworkedQuery(world).forEach(eid => {
+    const obj = world.eid2obj.get(eid);
+    if (obj) {
+      const textObj = world.eid2obj.get(renderAsEntity(world, NetworkDebugPrefab()))! as TroikaText;
+      texts.set(eid, textObj);
+      scene.add(textObj);
+      textObj.renderOrder = 999;
+      forEachMaterial(textObj, function (mat: Material) {
+        mat.depthTest = false;
+      });
+    }
+  });
+
+  exitedNetworkedQuery(world).forEach(eid => {
+    removeEntity(world, texts.get(eid)!.eid!);
+    texts.delete(eid);
+  });
+
+  networkedQuery(world).forEach(eid => {
+    const networkedObj = world.eid2obj.get(eid)!;
+    const textObj = texts.get(eid)!;
+
+    networkedObj.updateMatrices();
```

Comment:
Let me confirm my understanding. A networked object's translation/rotation/scale can be updated by remote peers but its matrix isn't updated yet because the scene graph will be updated in or right befor renderer.render() but this system needs updated world matrix so updating it here, correct?

Reply:
We call `updateMatrices` _anytime_ we access the `matrixWorld` or `matrix` of an `Object3D`. If it hasn't been updated, then very little work is done. This is part of the contract we follow because of the matrix optimizations we apply on top of threejs. (See commit description here: https://github.com/mozillareality/three.js/commit/b889d4e5617d95caa02064d6c8da5ae2b1f98e1a )

Reply:
Ah, ok. That is the one that skips the matrices update if it isn't dirty.